compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/draganddrop/WebDragAndDropManager.kt

```diff
@@ -194,4 +179,63 @@ private fun IntArray.toUint8ClampedArray(): Uint8ClampedArray {
     }
 
     return uint8ClampedArray
+}
+
+private class InternalStartTransferScope(
+    private val density: Density,
+    var transferData: DragAndDropTransferData?
+) : PlatformDragAndDropSource.StartTransferScope {
+    /**
+     * Context for an ongoing drag session initiated from Compose.
+     */
+    var ghostImage: HTMLCanvasElement? = null
+
+    fun isActive(): Boolean = transferData != null
+
+    override fun startDragAndDropTransfer(
```

Comment:
This function mixes two things:
1. working with rendering and bitmap/arrays conversion
2. working with DOM and setting the data

It's not a blocker, but it will be much easier to read if you can separate it into clearly named functions and the function itself will be like this:
```kt
val bitmap = drawToBitmap(...)
document.canvas.imageData = bitmap.data
```